src/main/java/seedu/address/model/person/Appointment.java

```diff
@@ -0,0 +1,69 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents a Person's appointment in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidAppointment(String)}
+ */
+public class Appointment {
+
+    public static final String MESSAGE_CONSTRAINTS = "Appointments should be of the format DD-MM-YYYY HH:MM";
+
+    public static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm");
+
+    public final LocalDateTime value;
+
+    /**
+     * Constructs an {@code Appointment}.
+     *
+     * @param dateTimeString A valid appointment date string.
+     */
+    public Appointment(String dateTimeString) {
+        requireNonNull(dateTimeString);
+        checkArgument(isValidAppointment(dateTimeString), MESSAGE_CONSTRAINTS);
+        this.value = LocalDateTime.parse(dateTimeString, FORMATTER);
+    }
+
+    /**
+     * Returns true if a given string is a valid appointment date.
+     */
+    public static boolean isValidAppointment(String test) {
```

Comment:
maybe this method can be in the appropriate parser instead?

Reply:
Agreed, moved to parser. thanks